src/lib/updateGrantsGrantees.js

```diff
@@ -139,6 +139,47 @@ export async function processFiles() {
       {
         updateOnDuplicate: ['number', 'status', 'startDate', 'endDate', 'updatedAt', 'programSpecialistName', 'programSpecialistEmail', 'grantSpecialistName', 'grantSpecialistEmail'],
       });
+
+    // Load and Process grant replacement data.
+    const grantReplacementsData = await fs.readFile('./temp/grant_award_replacement.xml');
+    const grantReplacementsJson = JSON.parse(toJson(grantReplacementsData));
+    const grantReplacements = grantReplacementsJson
+      .grant_award_replacements.grant_award_replacement;
+
+    // Loop all replaced grants.
+    let loopIterations = 0;
+    if (grantReplacements) {
+      if (Array.isArray(grantReplacements)) {
+        loopIterations = grantReplacements.length;
+      } else {
+        loopIterations = 1;
+      }
+    }
+
+    // eslint-disable-next-line max-len
+    // eslint-disable-next-line consistent-return
+    for (let index = 0; index < loopIterations; index += 1) {
+      const g = grantReplacements[index];
+      // Verify the grant being replaced exists.
+      const currentGrantId = parseInt(g.replaced_grant_award_id, 10);
+      // eslint-disable-next-line no-await-in-loop
+      const currentGrant = await Grant.findOne({ where: { id: currentGrantId } });
+      if (currentGrant) {
+        // Verify the replacement grant exists.
+        const replacementGrantId = parseInt(g.replacement_grant_award_id, 10);
+        // eslint-disable-next-line no-await-in-loop
+        const replacementGrant = await Grant.findOne({ where: { id: replacementGrantId } });
+        if (replacementGrant) {
+          // Set the old grant id.
+          if (replacementGrant.oldGrantId !== currentGrant.id) {
+            replacementGrant.oldGrantId = currentGrant.id;
+            // eslint-disable-next-line no-await-in-loop
+            await replacementGrant.save({ fields: ['oldGrantId'] });
```

Comment:
We probably want to gather up all these promises so we can do a `Promise.all(...)` on them all so we don't wait here for each promise to resolve.